docker-compose.mssql.yml

```diff
@@ -0,0 +1,60 @@
+# This file is part of Dependency-Track.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (c) Steve Springett. All Rights Reserved.
+version: "3"
+
+services:
+  apiserver:
+    depends_on:
```

Comment:
This probably needs a `service_healthy` to ensure it waits for mssql to pass its healthcheck before letting the API server loose on it. E.g.
```
  depends_on:
    mssql-init:
      condition: service_completed_successfully
```

And then mssql-init would do the following:
```
  depends_on:
    mssql:
      condition: service_healthy
```

And then mssql would need a healthcheck, which I won't guess at, but [here's an example](https://github.com/Microsoft/mssql-docker/issues/133#issuecomment-1484064993).

Reply:
Thanks for the hint! I was relying in the services just restarting if the service they depend on isn't ready yet. But this is much cleaner!

Reply:
No worries! The other thing you could look at is using [`extends`](https://docs.docker.com/compose/extends) for sharing definitions between files, and using [profiles](https://docs.docker.com/compose/profiles) to toggle the use of metrics. Might slightly simplify the commands - e.g. you could say:
 - `docker compose -f docker-compose.postgres.yml up -d` for Postgres
 - `docker compose -f docker-compose.mssql.yml --profile metrics up` for MS SQL with metrics
 - `docker-compose --profile metrics up` for H2 with metrics

Might be a bridge too far, but it wouldn't be hard to change if you fancied it.

Reply:
Will definitely look into it! I used profiles extensively in other projects but never mixed them with overrides.